src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -0,0 +1,946 @@
+// rekordboxfeature.cpp
+// Created 05/24/2019 by Evan Dekker
+
+#include <QMap>
+#include <QMessageBox>
+#include <QSettings>
+#include <QStandardPaths>
+#include <QtDebug>
+
+#include "library/rekordbox/rekordbox_pdb.h"
+#include "library/rekordbox/rekordboxfeature.h"
+
+#include "library/library.h"
+#include "library/librarytablemodel.h"
+#include "library/missingtablemodel.h"
+#include "library/queryutil.h"
+#include "library/trackcollection.h"
+#include "library/treeitem.h"
+#include "track/keyfactory.h"
+#include "util/db/dbconnectionpooled.h"
+#include "util/db/dbconnectionpooler.h"
+#include "util/file.h"
+#include "util/sandbox.h"
+#include "waveform/waveform.h"
+
+#include "widget/wlibrary.h"
+#include "widget/wlibrarytextbrowser.h"
+
+#define IS_RECORDBOX_DEVICE "::isRecordboxDevice::"
+#define IS_NOT_RECORDBOX_DEVICE "::isNotRecordboxDevice::"
+
+namespace {
+const QString kPDBPath = "PIONEER/rekordbox/export.pdb";
+const QString kPLaylistPathDelimiter = "-->";
+
+void clearTable(QSqlDatabase& database, QString tableName) {
+    QSqlQuery query(database);
+    query.prepare("delete from " + tableName);
+
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query)
+                << "tableName:" << tableName;
+    } else {
+        query.prepare("delete from sqlite_sequence where name='" + tableName + "'");
+
+        if (!query.exec()) {
+            LOG_FAILED_QUERY(query)
+                    << "tableName:" << tableName;
+        } else {
+            qDebug() << "Rekordbox table entries of '" << tableName << "' have been cleared.";
+        }
+    }
+}
+
+// This function is executed in a separate thread other than the main thread
+QList<TreeItem*> findRekordboxDevices(RekordboxFeature* rekordboxFeature) {
+    QThread* thisThread = QThread::currentThread();
+    thisThread->setPriority(QThread::LowPriority);
+
+    QList<TreeItem*> foundDevices;
+
+#if defined(__WINDOWS__)
+    // Repopulate drive list
+    QFileInfoList drives = QDir::drives();
```

Comment:
Unfortunately, no. The main rekordbox database format is not the same as this deviceSql format, which is designed to work in limited-power and limited-memory devices. I have only reverse-engineered the export format, because that is what I need in order to work with CDJs in performance settings. (Crate Digger was developed to solve the problem that [Beat Link Trigger](https://github.com/Deep-Symmetry/beat-link-trigger) was unable to obtain track metadata when a venue was set up with four CDJs because their network tables were full, preventing the direct database query approach from working.)

Reply:
Thank you for clarifying. This should be documented in a code comment somewhere.

Reply:
This is documented at the begining of https://github.com/ehendrikd/mixxx/blob/rekordbox-device-library-feature/src/library/rekordbox/rekordboxfeature.h